karapace/schema_models.py

```diff
@@ -37,15 +36,8 @@ def parse_avro_schema_definition(s: str, validate_enum_symbols: bool = True, val
     parsing fails because of it, the extra data can be removed and parsed
     again.
     """
-    try:
-        json_data = json.loads(s)
-    except json.JSONDecodeError as e:
-        if e.msg != "Extra data":
-            raise
-
-        json_data = json.loads(s[: e.pos])
-
```

Comment:
I removed this whole block as this accepts invalid JSON. If there is trailing data the data shall be considered bad.
For example `{"type": "fish"},["trailing"]` was splitted with the position from `json.JSONDecodeError` to `{"type": "fish"}`. The result is valid JSON, but should not be accepted as the input is garbage.